azure/src/main/scala/quasar/blobstore/azure/rx.scala

```diff
@@ -92,15 +92,16 @@ object rx {
   def enqueueEvent[F[_]: Effect, A](q: Queue[F, A])(event: A): Unit =
     Effect[F].runAsync(q.enqueue1(event))(_ => IO.unit).unsafeRunSync
 
-  def mkAsync[F[_], A](
+  def mkAsync[F[_]: ContextShift, A](
       observer: AsyncObserver[A], f: SingleObserver[A] => Unit)(
       implicit F: Async[F]): F[A] =
-    F.async[A] { cb: (Either[Throwable, A] => Unit) =>
-      observer.setCallback(cb)
-      f(observer)
-    }
+    F.guarantee(
+      F.async[A] { cb: (Either[Throwable, A] => Unit) =>
+        observer.setCallback(cb)
+        f(observer)
+      }) (ContextShift[F].shift)
```

Comment:
Note that I had to wrap `async` in `guarantee` rather than adding `ContextShift.shift` in the monad

Reply:
Similar change in `quasar-datasource-mongo` is coming up